pkg/report/model/cyclonedx.go

```diff
@@ -0,0 +1,261 @@
+	for i := range summary.Queries {
+		files := summary.Queries[i].Files
+		for idx := range files {
+			if !match(fileNames, files[idx].FileName) {
+				fileNames = append(fileNames, files[idx])
+			}
+		}
+	}
+	return fileNames
+}
+
+func generateSha256(filePath string) string {
+	content, err := os.ReadFile(filePath)
+
+	if err != nil {
+		log.Trace().Msgf("failed to read %s", filePath)
+		return ""
+	}
+
+	hashSum := sha256.Sum256(content)
+	return hex.EncodeToString(hashSum[:])
+}
+
+func getPurl(filePath, version string) string {
+	return fmt.Sprintf("pkg:generic/%s@%s", filePath, version)
+}
+
+func getDescription(query *model.QueryResult) string {
+	queryDescription := query.Description
+
+	fmt.Println(query.CISDescriptionTextFormatted)
+	if query.CISDescriptionTextFormatted != "" {
+		queryDescription = fmt.Sprintf("[CIS]: %s", query.CISDescriptionTextFormatted)
+	}
+
+	description := fmt.Sprintf("[%s].[%s]: %s", query.Platform, query.QueryName, queryDescription)
+
+	return description
+}
+
+func getVulnerabilities(fileName, purl string, summary *model.Summary) []Vulnerability {
+	var vulns []Vulnerability
+	for i := range summary.Queries {
+		query := summary.Queries[i]
+		for idx := range query.Files {
+			file := query.Files[idx]
+			if fileName == file.FileName {
+				vuln := Vulnerability{
+					Ref: purl + query.QueryID,
+					ID:  query.QueryID,
+					Source: Source{
+						Name: KICS,
+						URL:  "https://kics.io/",
+					},
+					Ratings: []Rating{
+						{
+							Severity: cycloneDxSeverityLevelEquivalence[query.Severity],
+							Method:   "Other",
+						},
+					},
+					Description: getDescription(&query),
+					Recommendations: []Recommendation{
+						{
+							Recommendation: fmt.Sprintf("In line %d, a result was found '%s', but '%s'",
+								file.Line, file.KeyActualValue, file.KeyExpectedValue),
+						},
+					},
+				}
+				vulns = append(vulns, vuln)
+			}
+		}
+	}
+	return vulns
+}
+
+// InitCycloneDxReport inits the CycloneDx report with no components (consequently, no vulnerabilities)
+func InitCycloneDxReport() *CycloneDxReport {
+	metadata := Metadata{
+		Timestamp: time.Now().Format(time.RFC3339),
+		Tools: &[]Tool{
+			{
+				Vendor:  "Checkmarx",
+				Name:    KICS,
+				Version: constants.Version,
+			},
+		},
+	}
+
+	return &CycloneDxReport{
+		XMLNS:        "http://cyclonedx.org/schema/bom/1.3",
+		XMLNSV:       "http://cyclonedx.org/schema/ext/vulnerability/1.0",
+		SerialNumber: "urn:uuid:" + uuid.New().String(),
+		Version:      1,
+		Metadata:     &metadata,
+	}
+}
+
+// BuildCycloneDxReport builds the CycloneDX report
+func BuildCycloneDxReport(summary *model.Summary) *CycloneDxReport {
+	var c Component
+	var vuln []Vulnerability
+	var version, sha, purl, filePath string
+
+	bom := InitCycloneDxReport()
+	files := getAllFiles(summary)
+
+	for i := range files {
+		filePath = strings.Replace(files[i].FileName, "\\", "/", -1)
+		sha = generateSha256(filePath)
+
+		index := 12
```

Comment:
I think we could define this as a constant outside the loop with some name like `minShaLength`